lenskit-core/src/main/java/org/lenskit/data/entities/Entity.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2014 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.data.entities;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Base class for data entities in LensKit.  The LensKit data model consists of various entities, each of which has a
+ * type, a numeric identifier (as a `long`), and additional data in a key-value mapping.
+ *
+ * Two entities are equal if their types, IDs, and data are equal.
+ *
+ * Identifiers are scoped per-type; it is acceptable to LensKit for two entities with different types to have the same
+ * identifier, and they are not considered to be the same entity.  However, individual data storage facilities are free
```

Comment:
Basically, you can implement a DAO where all entities have to have different IDs, regardless of type, if you feel like it, and you won't be breaking `Entity`'s contract.

Do I even need to mention that, or is it sufficiently clear that the fact that it's acceptable to LensKit does not imply it's acceptable to every application that might implement a LensKit DAO?